rgw/v2/tests/s3_swift/test_bucket_lc_object_exp_multipart.py

```diff
@@ -0,0 +1,239 @@
+                        object_name_downloaded = object_name + "." + "download"
+                        object_download_path = os.path.join(
+                            TEST_DATA_PATH, object_name_downloaded
+                        )
+                        object_downloaded_status = s3lib.resource_op(
+                            {
+                                "obj": bucket,
+                                "resource": "download_file",
+                                "args": [object_name, object_download_path],
+                            }
+                        )
+                        if object_downloaded_status is False:
+                            log.info("As expected object is not Downloadable")
+                        if object_downloaded_status is None:
+                            raise TestExecError(
+                                "Objects are not listed but can be downloadable"
+                            )
+
+                if config.local_file_delete is True:
+                    log.info("deleting local file created after the upload")
+                    utils.exec_shell_cmd(f"rm -rf {TEST_DATA_PATH}")
+
+                reusable.delete_bucket(bucket)
+        reusable.remove_user(each_user)
+
+    # check for any crashes during the execution
+    crash_info = reusable.check_for_crash()
+    if crash_info:
+        raise TestExecError("ceph daemon crash found!")
+
+
+if __name__ == "__main__":
+
+    test_info = AddTestInfo("bucket life cycle: test object expiration")
+    test_info.started_info()
+
+    try:
+        project_dir = os.path.abspath(os.path.join(__file__, "../../.."))
+        test_data_dir = "test_data"
+        TEST_DATA_PATH = os.path.join(project_dir, test_data_dir)
+        log.info("TEST_DATA_PATH: %s" % TEST_DATA_PATH)
+        if not os.path.exists(TEST_DATA_PATH):
+            log.info("test data dir not exists, creating.. ")
+            os.makedirs(TEST_DATA_PATH)
+        parser = argparse.ArgumentParser(description="RGW S3 Automation")
+        parser.add_argument("-c", dest="config", help="RGW Test yaml configuration")
+        parser.add_argument(
+            "-log_level",
+            dest="log_level",
+            help="Set Log Level [DEBUG, INFO, WARNING, ERROR, CRITICAL]",
+            default="info",
+        )
+        args = parser.parse_args()
+        yaml_file = args.config
+        log_f_name = os.path.basename(os.path.splitext(yaml_file)[0])
+        configure_logging(f_name=log_f_name, set_level=args.log_level.upper())
+        config = Config(yaml_file)
+        config.read()
+
+        if config.mapped_sizes is None:
+            config.mapped_sizes = utils.make_mapped_sizes(config)
+        test_exec(config)
+        test_info.success_status("test passed")
+        sys.exit(0)
+
+    except (RGWBaseException, Exception) as e:
+        log.info(e)
+        log.info(traceback.format_exc())
```

Comment:
We could simplify this by using `log.exception(e)`

Reply:
Made changes as per comments